TYPO3.Flow/Classes/TYPO3/Flow/Utility/Lock/FlockLockStrategy.php

```diff
@@ -51,56 +55,97 @@ class FlockLockStrategy implements LockStrategyInterface
     /**
      * @param string $subject
      * @param boolean $exclusiveLock TRUE to, acquire an exclusive (write) lock, FALSE for a shared (read) lock.
-     * @throws LockNotAcquiredException
-     * @throws \TYPO3\Flow\Utility\Exception
      * @return void
      */
     public function acquire($subject, $exclusiveLock)
     {
-        if ($this->isWindowsOS()) {
-            return;
-        }
         if (self::$temporaryDirectory === null) {
-            if (Bootstrap::$staticObjectManager === null || !Bootstrap::$staticObjectManager->isRegistered('TYPO3\Flow\Utility\Environment')) {
-                throw new LockNotAcquiredException('Environment object could not be accessed', 1386680952);
-            }
-            $environment = Bootstrap::$staticObjectManager->get('TYPO3\Flow\Utility\Environment');
-            $temporaryDirectory = Files::concatenatePaths(array($environment->getPathToTemporaryDirectory(), 'Lock'));
-            Files::createDirectoryRecursively($temporaryDirectory);
-            self::$temporaryDirectory = $temporaryDirectory;
+            $this->configureTemporaryDirectory();
         }
-        $this->lockFileName = Files::concatenatePaths(array(self::$temporaryDirectory, md5($subject)));
 
-        if (($this->filePointer = @fopen($this->lockFileName, 'r')) === false) {
-            if (($this->filePointer = @fopen($this->lockFileName, 'w')) === false) {
-                throw new LockNotAcquiredException(sprintf('Lock file "%s" could not be opened', $this->lockFileName), 1386520596);
-            }
+        $this->lockFileName = Files::concatenatePaths([self::$temporaryDirectory, md5($subject)]);
+        $continueTrying = true;
+        while ($continueTrying) {
+            $continueTrying = $this->tryToAquireLock($exclusiveLock);
+        }
+    }
+
+    /**
+     * Sets the temporaryDirectory as static variable for the lock class.
+     *
+     * @throws LockNotAcquiredException
+     * @throws \TYPO3\Flow\Utility\Exception
+     * return void;
+     */
+    protected function configureTemporaryDirectory()
+    {
+        if (Bootstrap::$staticObjectManager === null || !Bootstrap::$staticObjectManager->isRegistered(\TYPO3\Flow\Utility\Environment::class)) {
+            throw new LockNotAcquiredException('Environment object could not be accessed', 1386680952);
+        }
+        $environment = Bootstrap::$staticObjectManager->get(\TYPO3\Flow\Utility\Environment::class);
+        $temporaryDirectory = Files::concatenatePaths([$environment->getPathToTemporaryDirectory(), 'Lock']);
+        Files::createDirectoryRecursively($temporaryDirectory);
+        self::$temporaryDirectory = $temporaryDirectory;
+    }
+
+    /**
+     * Tries to open a lock file and apply the lock to it.
+     *
+     * @param boolean $exclusiveLock
+     * @return boolean True if a lock could not be aquired, false if the lock was successful.
```

Comment:
Agreed. Even if where this is used it sounds logical (`while ($continueTrying)`, it would be easier to understand here. And then when using it just do `while ($lockAcquired !== true)`.